aspnetcore/fundamentals/http-requests/samples/GitHub/RepoService.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace HttpClientFactorySample.GitHub
+{
+    #region snippet1
+    public class RepoService
+    {
+        private readonly HttpClient _httpClient; // not exposed publically
+
+        public RepoService(HttpClient client)
+        {
+            _httpClient = client;
+        }
+
+        public async Task<IEnumerable<string>> GetRepos()
+        {
+            var result = await _httpClient.GetAsync("aspnet/repos");
+
+            IEnumerable<string> resultObj = null;
+
+            if (result.IsSuccessStatusCode)
+            {
+                resultObj = JsonConvert.DeserializeObject<IEnumerable<string>>(await result.Content.ReadAsStringAsync()).ToList();
+            }
+
+            return resultObj ?? Array.Empty<string>();
```

Comment:
We should show throwing an exception rather than returning null or empty. Just use https://github.com/rynowak/Samples/blob/dev/MusicStore/src/MusicStoreUI/Services/MusicStoreService.cs#L35